packages/blade/src/components/BottomSheet/BottomSheetBody.web.tsx

```diff
@@ -20,12 +22,22 @@ const bodyStyles: React.CSSProperties = {
 const _BottomSheetBody = ({ children }: { children: React.ReactNode }): React.ReactElement => {
   const { scrollRef, setContentHeight, isOpen, bind } = useBottomSheetContext();
   const contentRef = React.useRef<HTMLDivElement>(null);
+  const [bottomSheetHasActionList, setBottomSheetHasActionList] = React.useState<boolean>(false);
 
   useIsomorphicLayoutEffect(() => {
     if (!contentRef.current) return;
     setContentHeight(contentRef.current.getBoundingClientRect().height);
   }, [contentRef, isOpen]);
 
+  React.useEffect(() => {
+    setBottomSheetHasActionList(false);
+    React.Children.forEach(children, (child) => {
+      if (isValidAllowedChildren(child, componentIds.ActionList)) {
+        setBottomSheetHasActionList(true);
+      }
+    });
```

Comment:
Hope no batching / non-batching issue with this block, otherwise we should set the bottom sheet has action list value at the end of this block (once).